pkg/adapter/metrics.go

```diff
@@ -69,7 +69,19 @@ type (
 		Builder
 
 		// NewMetrics returns a new instance of the Metrics aspect.
-		NewMetrics(env Env, config AspectConfig) (MetricsAspect, error)
+		NewMetrics(env Env, config AspectConfig, metrics []MetricDefinition) (MetricsAspect, error)
+	}
+
+	// MetricDefinition provides the basic description of a metric schema
+	// for which metrics adapters will be sent Values at runtime.
+	MetricDefinition struct {
+		// Name is the canonical name of the metric.
+		Name string
+		// Kind provides type information about the metric.
+		Kind Kind
+		// Labels are the names of keys for dimensional data that will
+		// generated at runtime and passed along with metric values.
+		Labels []string
```

Comment:
I thought about passing in a map to types briefly but had forgotten that we also had attribute descriptors and so skipped it. Thanks for the reminder!

I'll update to send a type map. I think i'll probably add a layer of indirection between the descriptor enum and what gets down, but yes, this seems like ultimately something we should support.